dask_ml/impute.py

```diff
@@ -80,9 +80,12 @@ def _fit_frame(self, X):
 
     def transform(self, X):
         if isinstance(X, (pd.Series, pd.DataFrame, dd.Series, dd.DataFrame)):
-            return X.fillna(self.statistics_)
+            if self.strategy == "mean" or self.strategy == "median":
+                return X.astype(np.number).fillna(self.statistics_)
+            else:
+                return X.fillna(self.statistics_)
```

Comment:
This `else` clause isn't necessary. The `return` statement above will always exit.

Reply:
I think this would be a little cleaner:

``` python
    def transform(self, X):
        if isinstance(X, (pd.Series, pd.DataFrame, dd.Series, dd.DataFrame)):
            if self.strategy == "mean" or self.strategy == "median":
                X = X.astype(np.number)
            return X.fillna(self.statistics_)
```

Reply:
I just added the change you suggested. 
The warnings present are in files I haven't touched, and are already present in the master branch